packages/features/form-builder/FormBuilder.tsx

```diff
@@ -475,65 +494,184 @@ export const FormBuilder = function FormBuilder({
+                      {fieldType?.isTextType ? (
+                        <InputField
+                          {...fieldForm.register("placeholder")}
+                          containerClassName="mt-6"
+                          label={t("placeholder")}
+                          placeholder={t(fieldForm.getValues("defaultPlaceholder") || "")}
+                        />
+                      ) : null}
+                      {fieldType?.needsOptions && !fieldForm.getValues("getOptionsAt") ? (
+                        <Controller
+                          name="options"
+                          render={({ field: { value, onChange } }) => {
+                            return <OptionsField onChange={onChange} value={value} className="mt-6" />;
+                          }}
+                        />
+                      ) : null}
+                      <Controller
+                        name="required"
+                        control={fieldForm.control}
+                        render={({ field: { value, onChange } }) => {
+                          return (
+                            <BooleanToggleGroupField
+                              data-testid="field-required"
+                              disabled={fieldForm.getValues("editable") === "system"}
+                              value={value}
+                              onValueChange={(val) => {
+                                onChange(val);
+                              }}
+                              label={t("required")}
+                            />
+                          );
+                        }}
+                      />
+                    </>
+                  );
+                }
+
+                if (!fieldTypeConfigVariantsConfig) {
+                  throw new Error("Field has `variantsConfig` but no `fieldTypeConfig`");
+                }
+
+                const variantToggleLabel = t(fieldTypeConfigVariantsConfig.toggleLabel || "");
+
+                if (!fieldType?.isTextType) {
+                  throw new Error("Variants are currently supported only with text type");
+                }
+
+                const defaultVariant = fieldTypeConfigVariantsConfig.defaultVariant;
+
+                const variants = Object.keys(variantsConfig.variants);
+                const otherVariants = variants.filter((v) => v !== defaultVariant);
+                if (otherVariants.length > 1) {
+                  throw new Error("More than one other variant. Remove toggleLabel ");
+                }
+                const otherVariant = otherVariants[0];
+                const variantName = fieldForm.watch("variant") || defaultVariant;
+                const variantFields =
+                  variantsConfig.variants[variantName as keyof typeof variantsConfig]?.fields;
+                const isSimpleVariant = variantFields.length === 1;
+                const isDefaultVariant = variantName === defaultVariant;
+                return (
```

Comment:
It's becoming too big, might want to think about creating a nested component for this.